contracts/anchor/OriginObserver.sol

```diff
@@ -0,0 +1,82 @@
+pragma solidity >=0.5.0 <0.6.0;
+
+// Copyright 2020 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import "../anchor/GenesisOriginObserver.sol";
+import "../anchor/StateRootProvider.sol";
+import "../consensus/CoconsensusModule.sol";
+import "../proxies/MasterCopyNonUpgradable.sol";
+
+/**
+ * @title OriginObserver - Contract on the auxiliary chain that stores the
+ *        state root of the origin chain.
+ *
+ * @notice This contract stores the state root and block height from the
+ *         origin chain. This contract is Coconsensus module. Only coconsensus
+ *         contract will be able to anchor the state root.
+ */
+contract OriginObserver is
+    MasterCopyNonUpgradable,
+    GenesisOriginObserver,
+    CoconsensusModule,
+    StateRootProvider
+{
+    /* Constants */
+
+    /** Storage index for genesisBlockNumber */
+    uint8 public constant genesisBlockNumberIndex = uint8(1);
```

Comment:
why would we store this? and if we store this, we should have a more general look up inside the genesis contract itself.

style guide note, constants should be capitalised SNAKE_CASE; but best just remove

Reply:
I was thinking that this may be helpful while creating the genesis file, I have removed this.
When we work on validators and genesis creation ticket, we can think at that time.